pkg/queue/concurrency_state.go

```diff
@@ -81,10 +81,48 @@ func ConcurrencyStateHandler(logger *zap.SugaredLogger, h http.Handler, pause, r
 			return
 		}
 
-		resume()
+		logger.Info("Requests increased from zero")
+		if err := resume(); err != nil {
+			logger.Errorf("Error handling resume request: %v", err)
+			panic(err)
+		}
 		paused = false
+		logger.Debug("From-Zero request successfully processed")
 		mux.Unlock()
 
 		h.ServeHTTP(w, r)
 	}
 }
+
+// ConcurrencyState Request sends to the concurrency state endpoint
+func ConcurrencyStateRequest(endpoint string, action ConcurrencyStateMessageBody) func() error {
+	bodyText, err := CreateConcurrencyStateMessageBody(action)
+	if err != nil {
+		_ = fmt.Errorf("unable to create message body: %w", err)
+		os.Exit(1)
```

Comment:
I don't think this does what you think it does. It won't log the message for instance.
I think we should surface the error through the return and `logger.Fatal` where this function is called.

Reply:
This may be a silly question, but given that this function returns a `func() error`, how would we return just the error here?